### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by ab-noori**
### 🌟Features:

> - [x] - The current_user method in ApplicationController has been removed, and authentication is now handled by Devise.
> - [x] - Devise has been successfully installed and configured, providing user registration and login functionality.
> - [x] - Users can now register using their email and password, with hashed passwords securely stored in the database.
> - [x] - Devise's email confirmation and password reset features have been implemented, enhancing account security and recovery options.

### CODE REVIEW

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @ab-noori , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- The current_user method in ApplicationController has been removed :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- The application allows for registering a new user :heavy_check_mark: 
- A user can log in with a combination of email and password :heavy_check_mark: 
- A hashed password is stored in the database  :heavy_check_mark: 

## Required Changes ♻️
- [ ] Great job so far @ab-noori There is just a small change I will suggest you implement and you'll be on your way to the next milestone :wink: 
- [ ] I noticed that your migration files were missing the devise migration file as a result when the project is cloned and the migrations are run, the columns that were supposed to be added by `devise` weren't included in your `user table` and as a result when I tried to check if your design matches the given wireframes, I got this error below. So kindly fix this so your app can run successfully without any errors :wink: 
- [ ] Also, kindly ensure that your design matches the given wireframe :wink: 
Screenshot from 2023-09-02 14-21-27

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ab-noori — COMMENTED** [review]
**Hi @NahnahAJ 👋,**
Thank you so much 🙏 for reviewing my pull request.
Your nice feedback is really appreciated 🙏.
I have checked and made them fix 🎯.

thank-you-gif-22

File: db/migrate/20230902155110_add_devise_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[7.0]
```
Comment (Whoistolu) [inline db/migrate/20230902155110_add_devise_to_users.rb:3]: - [ ] Well resolved 👍🏽 
I was able to migrate successfully, sign up, confirm my account, and sign in successfully 🥇 
Screenshot 2023-09-02 at 18 05 30

**Review by Whoistolu — APPROVED** [review]
### Hi @ab-noori ,

Great job implementing the recommended changes by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ab-noori** [comment]
**Hi @Whoistolu 👋,**
Thank you so much 🙏  for reviewing and approving my pull request.
Your nice feedback is really appreciated 🙏.

Thank-You-Very-Much

**Pull request merged**